File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.java
class Solution {
    public String kthDistinct(String[] v, int k) {
        HashMap<String,Integer> m=new HashMap<>();
        for(int i=0;i<v.length;i++){
            m.put(v[i],m.getOrDefault(v[i],0)+1);
        }
        for(int i=0;i<v.length;i++){
            if(m.get(v[i])==1){
                k--;
            }
            if(k==0){
                return v[i];
            }
        }
        return "";
    }
}

File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.cpp
class Solution {
public:
    string kthDistinct(vector<string>& v, int k) {
        map<string,int> m;
        string ans;
        for(auto q:v) m[q]++;
        for(int i=0;i<v.size();i++){
            if(m[v[i]]==1) k--;
            if(!k){
                ans=v[i];
                return ans;
            }
        }
        return ans;
    }
};